pints/_mcmc/__init__.py

```diff
@@ -543,39 +550,63 @@ def run(self):
 
             # Get points
             if self._single_chain:
-                xs = [sampler.ask() for sampler in self._samplers]
+                # Ask for log_pdf of points only in active samplers
+                xs = []
+                for i in i_active_samplers:
+                    xs.append(self._samplers[i].ask())
             else:
                 xs = self._samplers[0].ask()
 
-            # Calculate logpdfs
+            # Calculate logpdfs of points in active samplers
             fxs = evaluator.evaluate(xs)
 
             # Update evaluation count
-            evaluations += len(fxs)
```

Comment:
Is this chain necessary? The number of evaluations should equal the number of active samplers, no?